File: aat-android/src/main/java/ch/bailu/aat/util/ui/ToolTipView.java
package ch.bailu.aat.util.ui;


import android.content.Context;
import android.widget.TextView;

import ch.bailu.aat_lib.util.ui.ToolTipProvider;

public class ToolTipView extends TextView {
    public ToolTipView(Context context, UiTheme theme) {
        super(context);

        theme.toolTip(this);
        setVisibility(GONE);
    }

    public void setToolTip(ToolTipProvider tip) {

        String text = tip.getToolTip();

        if (text == null || text.length() == 0) {
            setVisibility(GONE);
        } else {
            setText(text);
            setVisibility(VISIBLE);
        }
    }
}

File: aat-android/src/main/java/ch/bailu/aat/util/ui/tooltip/ToolTipView.kt
package ch.bailu.aat.util.ui.tooltip

import android.content.Context
import android.widget.TextView
import ch.bailu.aat.util.ui.theme.UiTheme
import ch.bailu.aat_lib.util.ui.ToolTipProvider

class ToolTipView(context: Context, theme: UiTheme) : TextView(context) {
    init {
        theme.toolTip(this)
        visibility = GONE
    }

    fun setToolTip(tip: ToolTipProvider) {
        val text = tip.toolTip
        visibility = if (text == null || text.isEmpty()) {
            GONE
        } else {
            setText(text)
            VISIBLE
        }
    }
}
